tools/Import/Importer.cs

```diff
@@ -0,0 +1,358 @@
+                            {
+                                Console.WriteLine($"Retries={retries} Endpoint={endpoint} HttpStatusCode={statusString} ResourceType={resourceType} ResourceId={resourceId}");
+                            }
+
+                            if (statusString == "TooManyRequests") // retry overload errors forever
+                            {
+                                maxRetries++;
+                            }
+
+                            break;
+                    }
+                }
+                catch (Exception e)
+                {
+                    networkError = IsNetworkError(e);
+                    if (!networkError)
+                    {
+                        Console.WriteLine($"Retries={retries} Endpoint={endpoint} ResourceType={resourceType} ResourceId={resourceId} Error={(networkError ? "network" : e.Message)}");
+                    }
+
+                    bad = true;
+                    if (networkError) // retry network errors forever
+                    {
+                        maxRetries++;
+                    }
+                }
+                finally
+                {
+                    Interlocked.Decrement(ref epCalls);
+                }
+
+                if (bad && retries < maxRetries)
+                {
+                    retries++;
+                    Interlocked.Increment(ref waits);
+                    Thread.Sleep(networkError ? 1000 : 200 * retries);
+                    Interlocked.Decrement(ref waits);
+                }
+            }
+            while (bad && retries < maxRetries);
+            if (bad)
+            {
+                Console.WriteLine($"Failed writing ResourceType={resourceType} ResourceId={resourceId}. Retries={retries} Endpoint={endpoint}");
+            }
+        }
+
+        private static bool IsNetworkError(Exception e)
+        {
+            return e.Message.Contains("connection attempt failed", StringComparison.OrdinalIgnoreCase)
+                   || e.Message.Contains("connected host has failed to respond", StringComparison.OrdinalIgnoreCase)
+                   || e.Message.Contains("operation on a socket could not be performed", StringComparison.OrdinalIgnoreCase);
+        }
+
+        private static void ParseJson(string jsonString, out string resourceType, out string resourceId)
+        {
+            if (UseStringJsonParser)
+            {
+                var idStart = jsonString.IndexOf(@"""id"":""", StringComparison.OrdinalIgnoreCase) + 6;
+                var idShort = jsonString.Substring(idStart, 50);
+                var idEnd = idShort.IndexOf(@"""", StringComparison.OrdinalIgnoreCase);
+                resourceId = idShort.Substring(0, idEnd);
+                if (string.IsNullOrEmpty(resourceId))
+                {
+                    throw new ArgumentException("Cannot parse resource id with string parser");
+                }
+
+                var rtShort = jsonString.Substring(17, 30);
```

Comment:
Is this reliable?

Reply:
It is for a given file structure where resourcetype is first field. If it is last it will not work, so people can switch to full JObject type conversion.